index.ts

```diff
@@ -432,28 +446,236 @@ export class CapabilitiesController extends BaseController<any, any> implements
    * @param {string} domainName - The grantee domain.
    * @param {Array} newPermissions - The unique, new permissions for the grantee domain.
    */
-  addPermissionsFor (domainName: string, newPermissions: { [methodName: string]: IOcapLdCapability }) {
+  addPermissionsFor (
+    domainName: string,
+    newPermissions: { [methodName: string]: IOcapLdCapability }
+  ): void {
     const domain: RpcCapDomainEntry = this.getOrCreateDomainSettings(domainName);
     const newKeys = Object.keys(newPermissions);
 
-    // remove old permissions this will be overwritten
+    // remove old permissions so that they will be overwritten
     domain.permissions = domain.permissions.filter((oldPerm: IOcapLdCapability) => {
       return !newKeys.includes(oldPerm.parentCapability);
     });
 
     for (const methodName in newPermissions) {
-      const newPerm = newPermissions[methodName];
-      
-      domain.permissions.push(new Capability({
```

Comment:
Why is this call to `new Capability()` no longer needed? Was it not needed in the first place?

Reply:
It was not needed in the first place:
- `addPermissionsFor` takes an array of `IOcapLdCapability` objects
- Wherever we call `addPermissionsFor`, the capabilities have already been created using the constructor (at some point)